src/components/AddNote.vue

```diff
@@ -2,72 +2,120 @@
 import { ref } from 'vue'
 // import Component from "Component"
 
-
 export default {
 	name: "AddNote",
 	components: {
 		// Component,
 	},
 	props: { 
-		timestamp: Number,
+		timestamp: Date,
 	},
+	// emits: ['newNote'],
 	data() {
 		return {
-			timestampValue: "",
-			tags: [],
-			newTag: '',
-			entry: '',
+			hasError: false,
+			form: {
+				timestampValue: this.timestamp,
+				tags: '',
+				entry: '',
+			}
 		}
 	},
- methods: {
-		 addTag() {
-			  this.tags.push(this.newTag);
-			  this.newTag = '';
-		 },
-	},
-	mounted() {
-		this.timestampValue = this.timestamp.toString();
-	}
-}
-
+	methods: {
+		 submitAdd(event) {
+			 if (this.isValidDate() == true ) {
+				 this.$emit('newNote', this.form);
+			 } else {
+			 	this.hasError= true;
+			 }
 
+		 },
```

Comment:
There might a simpler way about validating a date, but I don't really know. Let's go with it for now.

FYI, you can validate the date input field when it changes, before form submit, by tying the validate function to the @input event on the field...
              @input="isValidDate"
And then just checking if there's an error in the submit.